pkg/env/integersetting.go

```diff
@@ -39,6 +42,12 @@ func (s *IntegerSetting) IntegerSetting() int {
 		return s.defaultValue
 	}
 	v, err := strconv.Atoi(val)
+	if slices.Contains(s.allowList, v) {
+		return v
+	}
```

Comment:
This will return 0 instead of the default value if `Atoi` failed, and 0 is in the allow list, right?

Reply:
That is a good catch actually! When atoi fails and 0 is on the allow list, then the default value should be returned. However, currently the 0 will be returned. I will add a regression test and patch that.